libs/tx-statemachine/src/error.test.ts

```diff
@@ -0,0 +1,111 @@
+import { ActionType, BuyAction } from '@thecointech/broker-db';
+import * as Transactions from '@thecointech/broker-db/transaction';
+import { GetContract } from '@thecointech/contract';
+import { getFirestore } from "@thecointech/firestore";
+import { init } from '@thecointech/firestore';
+import { log } from '@thecointech/logging';
+import Decimal from 'decimal.js-light';
+import { DateTime } from 'luxon';
+import * as FSM from '.';
+import { getCurrentState, StateGraph } from './types';
+import { manualOverride, manualOverrideTransition } from './transitions/manualOverride';
+
+const transitionBase = (type: string) =>({
+  timestamp: DateTime.now(),
+  type,
+  // Reset errors/meta
+  error: undefined,
+  meta: undefined,
+});
+// Simple transitions just test the different kind of scenarios the FSM needs to process.
+// noop - no change to data, just transition to a new state
+const noop = async () => transitionBase('noop');
+
+// Simulate an error occuring
+let shouldError = true;
+const maybeError = async () => ({
+  ...transitionBase("maybeError"),
+  ...(shouldError ? {error: "An error occurs" } : {})
+})
+
+type States = "initial"|"testError"|"error"|"complete";
+
+const graph : StateGraph<States, ActionType> = {
+  initial: {
+    // An error occurs when running the a transition
+    next: FSM.transitionTo<States>(maybeError, "testError"),
+  },
+  testError: {
+    next:  FSM.transitionTo<States>(noop, "complete"),
+    onError: FSM.transitionTo<States>(noop, "error"),
+  },
+  error: { next: manualOverride },
+  complete: null,
+}
+
+it("Error replay is handled appropriately", async () => {
```

Comment:
What is replay? The transaction?

Reply:
A replay event is one that has occurred before.

Our transactions are restart-able.  This allows them to handle situations like the congested networks gracefully, they just time-out and then restart when things die down again.  When restarting, they replay existing events to get the transaction back into the same state it was in before the error.

Reply:
I've just realized that our docs don't actually cover this anywhere